uPortal-content/uPortal-content-portlet/src/main/java/org/apereo/portal/portlet/dao/jpa/PortletDefinitionImpl.java

```diff
@@ -714,18 +665,59 @@ public EntityIdentifier getEntityIdentifier() {
 
     @Override
     public PortletLifecycleState getLifecycleState() {
+
+        /*
+         * NB:  A portlet definition may have a current lifecycle
+         * state of null if it has not yet been completely created.
+         */
+        PortletLifecycleState rslt = null;
+
         final Date now = new Date();
-        if (parameters.containsKey(PortletLifecycleState.MAINTENANCE_MODE_PARAMETER_NAME)) {
-            return PortletLifecycleState.MAINTENANCE;
-        } else if (expirationDate != null && expirationDate.before(now)) {
-            return PortletLifecycleState.EXPIRED;
-        } else if (publishDate != null && publishDate.before(now)) {
-            return PortletLifecycleState.PUBLISHED;
-        } else if (approvalDate != null && approvalDate.before(now)) {
-            return PortletLifecycleState.APPROVED;
-        } else {
-            return PortletLifecycleState.CREATED;
+        for (IPortletLifecycleEntry entry : lifecycleEntries) {
```

Comment:
@ChristianMurphy -- my own reading suggests that Streams are a terrific addition to Java 8 and will be a welcome tool in our uPortal developer toolkit, but also that they are not a "slam dunk" in every case.  They can be used in such a way where the performance is several times worse than the equivalent `for` loop (though it's not clear to me we would ever notice the difference, since the difference between -- for example -- 2 millis and 8 millis is only 6 millis).  Also, there are times when the Stream version of a loop is more clear (remarkably more clear) than the equivalent `for` loop, and there are circumstances where the opposite is true.

This is my vote on this subject (and many similar subjects):  no black-and-white rules about Streams.  If it possible to use a Stream, that **does not mean that using a Stream is a requirement** for merging.

This is an area where suggestions are very helpful, but rules are hurtful.